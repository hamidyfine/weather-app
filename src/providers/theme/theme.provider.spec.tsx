import { renderWrapper, screen } from '@/tests';

describe('ThemeProvider Component', () => {
    it('should render children', async () => {
        renderWrapper(
            <div>Children of Theme Provider</div>,
        );

        expect(screen.getByText('Children of Theme Provider')).toBeInTheDocument();
    });

    it('forwards props to MantineProvider', () => {
        const props = { theme: { defaultColorScheme: 'dark' } };

        renderWrapper(
            <div>Children of Theme Provider</div>,
            {
                ...props,
            },
        );

        const htmlElement = document.documentElement;

        expect(htmlElement).toHaveAttribute('data-mantine-color-scheme', 'dark');
    });

    it('renders correctly without additional props', () => {
        const { container } = renderWrapper(
            <div>Default Content</div>,
        );

        expect(container).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = renderWrapper(
            <div>Snapshot Content</div>,
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
