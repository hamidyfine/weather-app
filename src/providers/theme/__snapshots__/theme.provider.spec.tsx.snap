// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider Component matches snapshot 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-font-family: "Inter Variable", sans-serif;--mantine-font-family-headings: "Inter Variable", sans-serif;--mantine-color-dark-0: #E6E6E6;--mantine-color-dark-1: #CCCCCD;--mantine-color-dark-2: #B3B3B4;--mantine-color-dark-3: #99999B;--mantine-color-dark-4: #808081;--mantine-color-dark-5: #666668;--mantine-color-dark-6: #2A2C30;--mantine-color-dark-7: #222326;--mantine-color-dark-8: #19191C;--mantine-color-dark-9: #111112;--mantine-color-danger-0: #FDE3EC;--mantine-color-danger-1: #FAC8D4;--mantine-color-danger-2: #F7ADBC;--mantine-color-danger-3: #F492A4;--mantine-color-danger-4: #F1778C;--mantine-color-danger-5: #EE5C74;--mantine-color-danger-6: #EF3463;--mantine-color-danger-7: #C52A50;--mantine-color-danger-8: #9B213E;--mantine-color-danger-9: #70172C;--mantine-color-info-0: #E3F6F9;--mantine-color-info-1: #C8EEF3;--mantine-color-info-2: #ADE6ED;--mantine-color-info-3: #93DEE6;--mantine-color-info-4: #78D6E0;--mantine-color-info-5: #5DCEDA;--mantine-color-info-6: #42CBD8;--mantine-color-info-7: #35A3AD;--mantine-color-info-8: #287B82;--mantine-color-info-9: #1B5257;--mantine-color-light-0: #FFFFFF;--mantine-color-light-1: #FBFBFC;--mantine-color-light-2: #F8F9FB;--mantine-color-light-3: #F6F7FA;--mantine-color-light-4: #F5F6F9;--mantine-color-light-5: #F3F4F8;--mantine-color-light-6: #F1F3F7;--mantine-color-light-7: #C1C2C4;--mantine-color-light-8: #909294;--mantine-color-light-9: #606162;--mantine-color-primary-0: #EBEEFC;--mantine-color-primary-1: #CDD8F9;--mantine-color-primary-2: #AFC1F7;--mantine-color-primary-3: #91AAF4;--mantine-color-primary-4: #7394F1;--mantine-color-primary-5: #557DEE;--mantine-color-primary-6: #3068F4;--mantine-color-primary-7: #2853C2;--mantine-color-primary-8: #203E91;--mantine-color-primary-9: #182960;--mantine-color-secondary-0: #F0F2F8;--mantine-color-secondary-1: #E1E6F1;--mantine-color-secondary-2: #D3D9EB;--mantine-color-secondary-3: #C4CCE4;--mantine-color-secondary-4: #B6C0DE;--mantine-color-secondary-5: #A7B3D7;--mantine-color-secondary-6: #A8B5D2;--mantine-color-secondary-7: #868FAD;--mantine-color-secondary-8: #656889;--mantine-color-secondary-9: #434264;--mantine-color-success-0: #E1F4EF;--mantine-color-success-1: #C4E9DF;--mantine-color-success-2: #A6DDCE;--mantine-color-success-3: #89D2BE;--mantine-color-success-4: #6BC7AE;--mantine-color-success-5: #4EBB9D;--mantine-color-success-6: #0FA684;--mantine-color-success-7: #0C846A;--mantine-color-success-8: #096250;--mantine-color-success-9: #064036;--mantine-color-warning-0: #FDEFE1;--mantine-color-warning-1: #FBDDC3;--mantine-color-warning-2: #F9CAA6;--mantine-color-warning-3: #F7B889;--mantine-color-warning-4: #F5A56B;--mantine-color-warning-5: #F3934E;--mantine-color-warning-6: #F4A24D;--mantine-color-warning-7: #C3823E;--mantine-color-warning-8: #92612E;--mantine-color-warning-9: #61411F;}:root[data-mantine-color-scheme="dark"]{--mantine-color-dark-light: rgba(42, 44, 48, 0.15);--mantine-color-dark-light-hover: rgba(42, 44, 48, 0.2);--mantine-color-dark-outline-hover: rgba(128, 128, 129, 0.05);--mantine-color-danger-text: var(--mantine-color-danger-4);--mantine-color-danger-filled: var(--mantine-color-danger-8);--mantine-color-danger-filled-hover: var(--mantine-color-danger-9);--mantine-color-danger-light: rgba(239, 52, 99, 0.15);--mantine-color-danger-light-hover: rgba(239, 52, 99, 0.2);--mantine-color-danger-light-color: var(--mantine-color-danger-3);--mantine-color-danger-outline: var(--mantine-color-danger-4);--mantine-color-danger-outline-hover: rgba(241, 119, 140, 0.05);--mantine-color-info-text: var(--mantine-color-info-4);--mantine-color-info-filled: var(--mantine-color-info-8);--mantine-color-info-filled-hover: var(--mantine-color-info-9);--mantine-color-info-light: rgba(66, 203, 216, 0.15);--mantine-color-info-light-hover: rgba(66, 203, 216, 0.2);--mantine-color-info-light-color: var(--mantine-color-info-3);--mantine-color-info-outline: var(--mantine-color-info-4);--mantine-color-info-outline-hover: rgba(120, 214, 224, 0.05);--mantine-color-light-text: var(--mantine-color-light-4);--mantine-color-light-filled: var(--mantine-color-light-8);--mantine-color-light-filled-hover: var(--mantine-color-light-9);--mantine-color-light-light: rgba(241, 243, 247, 0.15);--mantine-color-light-light-hover: rgba(241, 243, 247, 0.2);--mantine-color-light-light-color: var(--mantine-color-light-3);--mantine-color-light-outline: var(--mantine-color-light-4);--mantine-color-light-outline-hover: rgba(245, 246, 249, 0.05);--mantine-color-primary-text: var(--mantine-color-primary-4);--mantine-color-primary-filled: var(--mantine-color-primary-8);--mantine-color-primary-filled-hover: var(--mantine-color-primary-9);--mantine-color-primary-light: rgba(48, 104, 244, 0.15);--mantine-color-primary-light-hover: rgba(48, 104, 244, 0.2);--mantine-color-primary-light-color: var(--mantine-color-primary-3);--mantine-color-primary-outline: var(--mantine-color-primary-4);--mantine-color-primary-outline-hover: rgba(115, 148, 241, 0.05);--mantine-color-secondary-text: var(--mantine-color-secondary-4);--mantine-color-secondary-filled: var(--mantine-color-secondary-8);--mantine-color-secondary-filled-hover: var(--mantine-color-secondary-9);--mantine-color-secondary-light: rgba(168, 181, 210, 0.15);--mantine-color-secondary-light-hover: rgba(168, 181, 210, 0.2);--mantine-color-secondary-light-color: var(--mantine-color-secondary-3);--mantine-color-secondary-outline: var(--mantine-color-secondary-4);--mantine-color-secondary-outline-hover: rgba(182, 192, 222, 0.05);--mantine-color-success-text: var(--mantine-color-success-4);--mantine-color-success-filled: var(--mantine-color-success-8);--mantine-color-success-filled-hover: var(--mantine-color-success-9);--mantine-color-success-light: rgba(15, 166, 132, 0.15);--mantine-color-success-light-hover: rgba(15, 166, 132, 0.2);--mantine-color-success-light-color: var(--mantine-color-success-3);--mantine-color-success-outline: var(--mantine-color-success-4);--mantine-color-success-outline-hover: rgba(107, 199, 174, 0.05);--mantine-color-warning-text: var(--mantine-color-warning-4);--mantine-color-warning-filled: var(--mantine-color-warning-8);--mantine-color-warning-filled-hover: var(--mantine-color-warning-9);--mantine-color-warning-light: rgba(244, 162, 77, 0.15);--mantine-color-warning-light-hover: rgba(244, 162, 77, 0.2);--mantine-color-warning-light-color: var(--mantine-color-warning-3);--mantine-color-warning-outline: var(--mantine-color-warning-4);--mantine-color-warning-outline-hover: rgba(245, 165, 107, 0.05);}:root[data-mantine-color-scheme="light"]{--mantine-color-dark-light: rgba(42, 44, 48, 0.1);--mantine-color-dark-light-hover: rgba(42, 44, 48, 0.12);--mantine-color-dark-outline-hover: rgba(42, 44, 48, 0.05);--mantine-color-danger-text: var(--mantine-color-danger-filled);--mantine-color-danger-filled: var(--mantine-color-danger-6);--mantine-color-danger-filled-hover: var(--mantine-color-danger-7);--mantine-color-danger-light: rgba(239, 52, 99, 0.1);--mantine-color-danger-light-hover: rgba(239, 52, 99, 0.12);--mantine-color-danger-light-color: var(--mantine-color-danger-6);--mantine-color-danger-outline: var(--mantine-color-danger-6);--mantine-color-danger-outline-hover: rgba(239, 52, 99, 0.05);--mantine-color-info-text: var(--mantine-color-info-filled);--mantine-color-info-filled: var(--mantine-color-info-6);--mantine-color-info-filled-hover: var(--mantine-color-info-7);--mantine-color-info-light: rgba(66, 203, 216, 0.1);--mantine-color-info-light-hover: rgba(66, 203, 216, 0.12);--mantine-color-info-light-color: var(--mantine-color-info-6);--mantine-color-info-outline: var(--mantine-color-info-6);--mantine-color-info-outline-hover: rgba(66, 203, 216, 0.05);--mantine-color-light-text: var(--mantine-color-light-filled);--mantine-color-light-filled: var(--mantine-color-light-6);--mantine-color-light-filled-hover: var(--mantine-color-light-7);--mantine-color-light-light: rgba(241, 243, 247, 0.1);--mantine-color-light-light-hover: rgba(241, 243, 247, 0.12);--mantine-color-light-light-color: var(--mantine-color-light-6);--mantine-color-light-outline: var(--mantine-color-light-6);--mantine-color-light-outline-hover: rgba(241, 243, 247, 0.05);--mantine-color-primary-text: var(--mantine-color-primary-filled);--mantine-color-primary-filled: var(--mantine-color-primary-6);--mantine-color-primary-filled-hover: var(--mantine-color-primary-7);--mantine-color-primary-light: rgba(48, 104, 244, 0.1);--mantine-color-primary-light-hover: rgba(48, 104, 244, 0.12);--mantine-color-primary-light-color: var(--mantine-color-primary-6);--mantine-color-primary-outline: var(--mantine-color-primary-6);--mantine-color-primary-outline-hover: rgba(48, 104, 244, 0.05);--mantine-color-secondary-text: var(--mantine-color-secondary-filled);--mantine-color-secondary-filled: var(--mantine-color-secondary-6);--mantine-color-secondary-filled-hover: var(--mantine-color-secondary-7);--mantine-color-secondary-light: rgba(168, 181, 210, 0.1);--mantine-color-secondary-light-hover: rgba(168, 181, 210, 0.12);--mantine-color-secondary-light-color: var(--mantine-color-secondary-6);--mantine-color-secondary-outline: var(--mantine-color-secondary-6);--mantine-color-secondary-outline-hover: rgba(168, 181, 210, 0.05);--mantine-color-success-text: var(--mantine-color-success-filled);--mantine-color-success-filled: var(--mantine-color-success-6);--mantine-color-success-filled-hover: var(--mantine-color-success-7);--mantine-color-success-light: rgba(15, 166, 132, 0.1);--mantine-color-success-light-hover: rgba(15, 166, 132, 0.12);--mantine-color-success-light-color: var(--mantine-color-success-6);--mantine-color-success-outline: var(--mantine-color-success-6);--mantine-color-success-outline-hover: rgba(15, 166, 132, 0.05);--mantine-color-warning-text: var(--mantine-color-warning-filled);--mantine-color-warning-filled: var(--mantine-color-warning-6);--mantine-color-warning-filled-hover: var(--mantine-color-warning-7);--mantine-color-warning-light: rgba(244, 162, 77, 0.1);--mantine-color-warning-light-hover: rgba(244, 162, 77, 0.12);--mantine-color-warning-light-color: var(--mantine-color-warning-6);--mantine-color-warning-outline: var(--mantine-color-warning-6);--mantine-color-warning-outline-hover: rgba(244, 162, 77, 0.05);}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div>
    Snapshot Content
  </div>
</DocumentFragment>
`;
